spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionController.java

```diff
@@ -205,12 +212,40 @@ public TaskExecutionResource view(@PathVariable("id") long id) {
 	 */
 	@RequestMapping(value = "/{id}", method = RequestMethod.DELETE)
 	@ResponseStatus(HttpStatus.OK)
-	public void cleanup(@PathVariable("id") long id) {
-		TaskExecution taskExecution = this.explorer.getTaskExecution(id);
-		if (taskExecution == null) {
-			throw new NoSuchTaskExecutionException(id);
+	public void cleanup(
+			@PathVariable("id") Set<Long> ids,
+			@RequestParam(defaultValue = "CLEANUP", name="action") TaskExecutionControllerDeleteAction[] actions) {
+
+		final Set<TaskExecutionControllerDeleteAction> actionsAsSet = new HashSet<>(Arrays.asList(actions));
+		final SortedSet<Long> nonExistingTaskExecutions = new TreeSet<>();
+
+		// Should move to service to ensure TX
```

Comment:
Is this a TODO or something that is immediate?

Reply:
Done